File: mawhub/app/job/dto/job_applicant.py
from typing import List, TypedDict

from mawhub.app.job.dto.applicant_resume import ApplicantResumeDTO

class JobApplicantCreateWithResume(TypedDict):
    applicant_resume: ApplicantResumeDTO
    pipeline_step_id: str
    job_opening_id: str

class JobApplicantUpdateRequest(TypedDict):
    name: str
    status: str
    pipeline_step: str

class JobApplicantBulkUpdateRequest(TypedDict):
    names: List[str]
    status: str
    pipeline_step: str

File: ats/src/tsgen/job_applicant.ts
export interface JobApplicantCreateWithResume {
  applicant_resume: ApplicantResumeDTO;
  pipeline_step_id: string;
  job_opening_id: string;
}

export interface JobApplicantUpdateRequest {
  name: string;
  status: string;
  pipeline_step: string;
}

export interface JobApplicantBulkUpdateRequest {
  names: string[];
  status: string;
  pipeline_step: string;
}
